src/pieces/utils.py

```diff
@@ -0,0 +1,50 @@
+import json
+from bs4 import BeautifulSoup
+import os
+import re
+
+from pieces.settings import Settings
+
+# Used to create a valid file name when opening to "Opened Snippets"
+def sanitize_filename(name):
+    """ Sanitize the filename by removing or replacing invalid characters. """
+    # Replace spaces with underscores
+    name = name.replace(" ", "_")
+    # Remove invalid characters
+    name = re.sub(r'[\\/*?:"<>|]', '', name)
+    return name
+
+def extract_code_from_markdown(markdown, name, language):
+    # Sanitize the name to ensure it's a valid filename
+    filename = sanitize_filename(name)
+    file_extension = get_file_extension(language)
+
+    # Using BeautifulSoup to parse the HTML and extract text
+    soup = BeautifulSoup(markdown, 'html.parser')
+    extracted_code = soup.get_text()
+
+    # Minimize multiple consecutive newlines to a single newline
+    extracted_code = re.sub(r'\n\s*\n', '\n', extracted_code)
```

Comment:
will need to probably also account for crlf here

Reply:
I think this function is useless not sure why it is added in the first place
lemme check jayson